### ISSUE TITLE
E_ACCESSDENIED when trying to install new distro

### ISSUE THREAD

**Issue by RequiDev** [microsoft/WSL#9326]
### Version

Microsoft Windows [Version 10.0.19045.2311]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.74.2

### Repro Steps

Anytime I try to install a distro, for example, Ubuntu it fails with:
Installing: Ubuntu
Access is denied. =========90.0%=====================      ]
Error code: Wsl/InstallDistro/E_ACCESSDENIED

If I do it through the Microsoft Store I get:
Code: 0x80070005

I removed an Ubuntu instance before this started happening because I used it as a temporary instance to tinker around.

### Expected Behavior

The distro should install.

### Actual Behavior

Installing: Ubuntu
Access is denied. =========90.0%=====================      ]
Error code: Wsl/InstallDistro/E_ACCESSDENIED

If I do it through the Microsoft Store I get:
Code: 0x80070005

### Diagnostic Logs

[WslLogs-2022-12-14_12-39-19.zip](https://github.com/microsoft/WSL/files/10227414/WslLogs-2022-12-14_12-39-19.zip)

**Comment by Shadow-Cipher** [comment]
Same issue. As a matter of fact, on three different physical computers using a couple of different versions of Windows Pro, there hasn't been a single time when I've installed WSL and used it heavily where it hasn't stopped working for some reason or another and usually a fix isn't easy.

**Comment by owen118z** [comment]
+1

**Comment by RequiDev** [comment]
I managed to fix this. The underlying problem was that the Microsoft Store was unable to install anything it seems.
I have fixed this by following the third solution here: https://answers.microsoft.com/en-us/windows/forum/all/0x80070005-cant-download-or-update-any-app-from/1629e23b-6332-4cb0-ad12-1b1fc12e01a1

In case this link is ever dead:
Solution 3
Reinstall the Microsoft Store.

1. Right-click on Start.
2. Click on Windows PowerShell(admin)
3. Type: Get-AppXPackage -allusers | Foreach {Add-AppxPackage -DisableDevelopmentMode -Register "$($_.InstallLocation)\AppXManifest.xml"} and press ENTER.
5. Reboot your pc and try open the Microsoft Store.
